### ISSUE TITLE
[bug] in AHK or Debug adapter or vscode?

### ISSUE THREAD

**Issue by SAbboushi** [zero-plusplus/vscode-autohotkey-debug#24]
I don't understand what is happening with this script.
The problems I experience:
(1) "Critical Error:  Function recursion limit exceeded." Only when "Global" variables is expanded in Debug Side Bar
(2) Port used for debugging is not released after this error, so new debug session assigns another port

Script:
```
#SingleInstance 
class ClassName {
    i := 0
    banana {
        Get{
            sleep 1
        }
    }
}
x:=ClassName.new()

/*
Critical Error:  Function recursion limit exceeded.

	Line#
	003: {
	000: }
	004: {
	000: Super.__Init()
	004: this.i := 0
	000: }
	006: {
--->	007: sleep(1)
	008: }
	012: x := ClassName.new()
	013: Exit

The program is now unstable and will exit.
```

Details and steps to recreate using AHK [a118]:
A)
- Set breakpoint on line 10
- Run script
- Note that "Global" is expanded
- Note that port is 9021

![image](https://user-images.githubusercontent.com/1964198/88100538-44d0f580-cb62-11ea-99d2-ceccd1773645.png)

B)
- Press F11 key ("Step Into")
- Execution pauses on line 3
- Critical Error:  Function recursion limit exceeded
- Note busy icon for Global 
![image](https://user-images.githubusercontent.com/1964198/88100835-bb6df300-cb62-11ea-8b8a-0631d867189b.png)

![image](https://user-images.githubusercontent.com/1964198/88101338-70a0ab00-cb63-11ea-97a7-fcf96125300b.png)

C)
- Click "OK" button
- Debug session terminates
- Note messages in Debug Console:
```
Session closed for the following reasons: read ECONNRESET
Session closed for the following reasons: Cannot call write after a stream was destroyed
```
![image](https://user-images.githubusercontent.com/1964198/88101652-e9a00280-cb63-11ea-8da3-03430007f3b3.png)

D)

- Press "F5" key ("Start Debugging")
- Prompted to use new port 9022 because 9021 was not released

![image](https://user-images.githubusercontent.com/1964198/88101952-6b902b80-cb64-11ea-896b-83b1c4fc99ac.png)

E)
- Click "Yes" for port 9022
- Collapse Global variables pane
![image](https://user-images.githubusercontent.com/1964198/88102622-7b5c3f80-cb65-11ea-9345-5ed162eafa06.png)

 F)
- Press F11 key ("Step Into")
- Execution pauses on line 3
- Now there is no recursion error:
![image](https://user-images.githubusercontent.com/1964198/88102878-de4dd680-cb65-11ea-82de-38d076252bc7.png)

G)
Exit vscode and relaunch vscode.  Now the ports can be re-used (9000, 9001...)

**Comment by zero-plusplus** [comment]
In conclusion, I think it's my problem.

Apparently, in AutoHotkeyV2, when the debugger inspects a property (in this case banana), the get-accessor is executed and tries to display its value. (This seems to be common in the Node debugger as well)

You can easily see this phenomenon by changing `Sleep 1` to `MsgBox`.

I'm not optimizing data-inspect, so I'm getting the same value multiple times. So there must have been a process somewhere that kept referring to banana. As a result, it seems to have crashed.

By the way, SciTE4AutoHotkey was designed to not crash.

This issue will be fixed in the next version, but the optimization process may take some time.

**Comment by zero-plusplus** [comment]
I thought it would take a long time, but I only had to add one line to solve it.

I've added it before, but for some reason I had a problem getting the child element, so I didn't think too much about it and removed it. (This is why I was getting the same value multiple times.)

This one line should be correct as a program, so there shouldn't be a bug, but I hadn't looked into the cause of the bug that prevented me from getting the child elements, so I'm a little concerned.

I'll test it as best I can, but I should mention that another bug may occur.
